esp-hal/src/spi/slave.rs

```diff
@@ -306,13 +264,28 @@ pub mod dma {
         }
     }
 
-    impl<'d, T, C, DmaMode> SpiDma<'d, T, C, DmaMode>
+    impl<'d, T, DmaMode> SpiDma<'d, T, DmaMode>
     where
         T: InstanceDma,
-        C: DmaChannel,
-        C::P: SpiPeripheral,
         DmaMode: Mode,
     {
+        fn new<CH>(
+            spi: PeripheralRef<'d, T>,
+            channel: Channel<'d, CH, DmaMode>,
+            rx_descriptors: &'static mut [DmaDescriptor],
+            tx_descriptors: &'static mut [DmaDescriptor],
+        ) -> Self
+        where
+            CH: DmaChannelConvert<T::Dma>,
+        {
+            channel.runtime_ensure_compatible(&spi);
```

Comment:
Is this still needed or are you keeping it around for peripheral erasure?

Reply:
I expect this to be still necessary in the future, yes-